libsel4/include/sel4/log.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <sel4/simple_types.h>
+#include <sel4/constants.h>
+#include <sel4/sel4_arch/constants.h>
+#include <sel4/benchmark_track_types.h>
+
+/*
+ * Log event implementation
+ * ========================
+ *
+ * Each event is a sequence of words in the log buffer. The first word
+ * of an event is the seL4_LogEvent which describes the length in words
+ * of the event and the event type. There is also an additional
+ * component to the seL4_LogEvent with use dependend on the event type.
+ *
+ * For an event of type t, there will be an event identifier
+ * seL4_Log_TypeId(t), and a struct representing the event
+ * seL4_Log_Type(t).
+ *
+ * The event can be logged from within the kernel using
+ * seL4_Log(t)(event_args...). seL4_Log expressions are automatically
+ * removed if logging is disabled.
+ */
+
+/* The event header word which appears at the start of every event */
+typedef struct {
+    /* Type of event logged */
+    seL4_Word type: 8;
+    /* The remaining 24 or 56 bits of the header word can be used for
+     * the event itself */
+#if seL4_WordSizeBits == 2
+    seL4_Word data: 24;
+#elif seL4_WordSizeBits == 3
+    seL4_Word data: 56;
+#else
+#error "Unsupported word seL4_WordSizeBits"
+#endif
+} seL4_LogEvent;
+
+/* The name of the type identifier for a given log event type */
+#define seL4_Log_TypeId(event) seL4_Log_TypeId_ ## event
```

Comment:
I might be a  bit rusty but I think the style guide still expects macros to be all caps?

Reply:
> Use `SCREAMING_SNAKE_CASE` for, and only for, preprocessor symbols and values of enumeration type.

Yup it does. Should make this look a bit less magical too.

* [ ] `SCREAMING_SNAKE_CASE` for preprocessor symbols